### ISSUE TITLE
Inconsistency between `$empty[1]` and `()[1]`

### ISSUE THREAD

**Issue by imgx64** [fish-shell/fish-shell#3177]
```
> set -le empty
> count $empty[1]
0
> count ()[1]
Array index out of bounds
fish: count ()[1]
               ^
```

I think the second should also be 0.

Oddly enough, there's a [test](https://github.com/fish-shell/fish-shell/blob/44cde9e0e992c1003eb019bb0fe68882b82fa217/tests/expansion.in#L81) for it and it [checks it's an error](https://github.com/fish-shell/fish-shell/blob/44cde9e0e992c1003eb019bb0fe68882b82fa217/tests/expansion.err#L25). Is there a reason for wanting this behavior?

**Comment by faho** [comment]
Compare #826, which is about $var[2] failing.

Personally I believe that ()[x] should just expand to nothing if x is invalid, even more than $var[x]. That's because you might have run a program in the command substitution that might not print the same output if run twice, so you almost always need to do `set var (); if set -q var[x]`.

Special-casing 1 like for variables is a workaround, not a real solution.

**Comment by krader1961** [comment]
Closing this as a dup of #826 since there seems to be consensus that we don't really want the special case of `$var[1]` for empty or missing vars. Which means we certainly don't want to extend that behavior to command substitution. You can get the behavior you desire, albeit using more verbose syntax, by doing something along the lines of

```
set -l tmpvar (some_command)
set -q tmpvar[1]
or echo no output
```

Once we have python slice semantics implemented per the most recent comments in #826 you'll be able to just do `(some_command)[1..1]`.
